controllers/mirrorpeersecret_controller.go

```diff
@@ -0,0 +1,282 @@
+		Name:      nPeerRef.StorageClusterRef.Name,
+		Namespace: nPeerRef.StorageClusterRef.Namespace,
+	}
+	var retSecret *corev1.Secret
+	if secretLabelType == secretutils.DestinationLabel {
+		retSecret = secretutils.CreateDestinationSecret(secretNamespacedName, scNamespacedName, nPeerRef.Data)
+	} else if secretLabelType == secretutils.SourceLabel {
+		retSecret = secretutils.CreateSourceSecret(secretNamespacedName, scNamespacedName, nPeerRef.Data)
+	}
+
+	return retSecret
+}
+
+// updateDestinationSecretsFromSource updates all destination secrets from source
+func (r *MirrorPeerSecretReconciler) updateDestinationSecretsFromSource(ctx context.Context, sourceSecret *corev1.Secret) error {
+	logger := log.FromContext(ctx)
+	if err := secretutils.ValidateSourceSecret(sourceSecret); err != nil {
+		logger.Error(err, "Updating the destination secrets failed. Invalid source secret.", "controller", "MirrorPeerSecret", "secret", sourceSecret)
+		return err
+	}
+	var mirrorPeerListObj multiclusterv1alpha1.MirrorPeerList
+	err := r.List(ctx, &mirrorPeerListObj)
+	if err != nil {
+		logger.Error(err, "Unable to get the list of MirrorPeer objects", "controller", "MirrorPeerSecret")
+		return err
+	}
+	logger.V(2).Info("Successfully got the list of MirrorPeers", "controller", "MirrorPeerSecret", "MirrorPeerListObj", mirrorPeerListObj)
+
+	uniqueConnectedPeers := ConnectedPeers(sourceSecret, mirrorPeerListObj.Items)
+	logger.Info("Listing all the Peers connected to the Source", "SourceSecret", sourceSecret, "AllUniqueConnectedPeers", uniqueConnectedPeers)
+
+	// anyErr will have the last found error
+	var anyErr error
+	for _, eachConnectedPeer := range uniqueConnectedPeers {
+		var nPeerRef NamedPeerRefWithSecretData = NamedPeerRefWithSecretData{
+			Name:    sourceSecret.Name,
+			Data:    sourceSecret.Data[secretutils.SecretDataKey],
+			PeerRef: eachConnectedPeer,
+		}
+		err := r.updateDestinationSecretFromPeerRef(ctx, nPeerRef)
+		if err != nil {
+			logger.Error(err, "Unable to update the destination secret", "controller", "MirrorPeerSecret", "PeerRef", eachConnectedPeer)
+			anyErr = err
+		}
+	}
+
+	return anyErr
+}
+
+func (r *MirrorPeerSecretReconciler) updateDestinationSecretFromPeerRef(ctx context.Context, peerRef NamedPeerRefWithSecretData) error {
+	logger := log.FromContext(ctx)
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      peerRef.Name,
+			Namespace: peerRef.ClusterName,
+		},
+	}
+
+	expectedDest := peerRef.GenerateSecret(secretutils.DestinationLabel)
+	var currentDest corev1.Secret
+	err := r.Get(ctx, req.NamespacedName, &currentDest)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("Creating destination secret", "controller", "MirrorPeerSecret", "secret", *expectedDest)
+			err = r.Create(ctx, expectedDest)
+		}
+		if err != nil {
+			logger.Error(err, "Unable to update the destination secret", "controller", "MirrorPeerSecret", "destination-ref", peerRef)
+		}
```

Comment:
```suggestion
		if errors.IsNotFound(err) {
			logger.Info("Creating destination secret", "controller", "MirrorPeerSecret", "secret", *expectedDest)
			err = r.Create(ctx, expectedDest)
			if err != nil {
			return err
			}
			
		}
	}
		
```
You are already doing the err check in the parent func when looping through peerRefs